pkg/api/types.go

```diff
@@ -498,6 +498,33 @@ type EnvVar struct {
 	Name string `json:"name"`
 	// Optional: defaults to "".
 	Value string `json:"value,omitempty"`
+	// Optional: specify a source for the value of the EnvVar.
+	*EnvVarSource `json:"source,omitempty"`
```

Comment:
What if user specifies Value and one of the alt sources?  Who wins?  Document it.